### PR TITLE
Allow users to turn off "Related posts" in Reading Settings

### PR DESCRIPTION
**Pull request by okmttdhr**
## Proposed changes:

This PR proposes allowing users to turn off the "Related posts" feature regardless of permission to toggle this feature. 

Please refer to https://github.com/Automattic/dotcom-forge/issues/6526#issuecomment-2124064612 for a detailed background.

Users who have permission to toggle this feature: 
| before | after |
|--------|--------|
| Screenshot 2024-05-23 at 10 06 57 | Screenshot 2024-05-23 at 10 06 09 | 

Users who do **not** have permission to toggle this feature (No change): 
| before | after |
|--------|--------|
| Screenshot 2024-05-23 at 10 06 09 | Screenshot 2024-05-23 at 10 06 09 | 

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Go to Settings > Reading
* Confirm "Related posts > 'Hide related content after posts'" is shown
* Confirm "Related posts > 'Hide related content after posts'" can be saved

### CODE REVIEW

**Review by DustyReagan — COMMENTED** [review]
I noticed a couple things:

On Atomic, if you disable Related posts in Jetpack Settings, the Related posts option in wp-admin completely disappears.

On Atomic, If you disable the Related posts is wp-admin, nothing changes for Related posts in Jetpack settings.

Nit: On my Simple Classic site, I noticed the checkboxes are indented, but not on Atomic.


https://github.com/Automattic/jetpack/assets/140841/02e26301-0746-4e45-aa9d-ac9bdbfc6f18

**Comment by okmttdhr** [comment]
>On Atomic, if you disable Related posts in Jetpack Settings, the Related posts option in wp-admin completely disappears.
On Atomic, If you disable the Related posts is wp-admin, nothing changes for Related posts in Jetpack settings.

Thank you for your comment, @DustyReagan! After taking a look at it again, this is because Jetpack Setting toggles the feature itself while *“Hide related content after posts”* in wp-admin changes the setting to hide it. 

TLDR: I’m leaning toward keeping the current behavior since it is consistent for users and feels pragmatic in terms of Untangling project. 

---
- This PR’s proposal
    - Structure
        - Atomic Classic
            - Jetpack Setting
                - Show related content after posts (This toggles the feature itself)
                    - options
            - wp-admin
                - 🆕 **Hide related content after posts (This does not toggle the feature itself, but users can hide it)**
                - Show related content after posts
                    - options
        - Simple Classic
            - ~~Jetpack Setting~~  (Users cannot access here)
            - wp-admin
                - Hide related content after posts (This does not toggle the feature itself, but users can hide it)
                - Show related content after posts
                    - options
    - Since Jetpack Setting toggles the feature itself, selecting *“Hide related content after posts”* in wp-admin does not get reflected in Jetpack Setting (i.e. the Related posts feature itself is still ON). This might confuse users when, for example, users try to turn on the feature from Jetpack Setting, but it does not show the related posts because *“Hide related content after posts”* is selected in wp-admin before.
- What if fully implement it to address inconsistencies that we have in this PR’s proposal? (by separating the feature toggle and the hide setting)
    - Structure
        - Jetpack Setting
            - 🆕 **Enable Related posts**
                - Show related content after posts
                    - options
        - wp-admin
            - Hide related content after posts
            - Show related content after posts
                - options
    - While it's consistent between Jetpack Setting and wp-admin, I’m not sure if it makes sense for users to turn off *“Show related content after posts”* in Jetpack Setting while enabling the feature. Users may just disable the feature. So, I also do not know if it’s worth implementing, as it may confuse users in other ways.
- Keep the current behavior
    - Structure
        - Atomic Classic
            - Jetpack Setting
                - Show related content after posts (This toggles the feature itself)
                    - options
            - wp-admin
                - Show related content after posts
                    - options
        - Simple Classic
            - ~~Jetpack Setting~~  (Users cannot access here)
            - wp-admin
                - Hide related content after posts (This does not toggle the feature itself, but users can hide it)
                - Show related content after posts
                    - options
    - This raises the differences between Atomic and Simple. However, it makes sense in that the feature is offered by default even on Simple, but can also be turned off.
    - There is another problem where users do not have a way to hide the related posts from Reading Settings, though.

more context: https://github.com/Automattic/dotcom-forge/issues/6526#issuecomment-2124064612
